### PR TITLE
Added initial state to StateMachineInfo, and to GetInfo of StateMachi…

### PR DESCRIPTION
**Pull request by HenningNT**
…ne. Added initial state line in digraph output.

As requested in #285

### CODE REVIEW

File: src/Stateless/Reflection/StateMachineInfo.cs [hunk 1]
```diff
@@ -20,6 +21,11 @@ internal StateMachineInfo(IEnumerable<StateInfo> states, Type stateType, Type tr
         /// Exposes the states, transitions, and actions of this machine.
```
Comment (ike86) [inline src/Stateless/Reflection/StateMachineInfo.cs]: You've missed this comment :wink:

File: src/Stateless/Graph/StateGraph.cs [hunk 2]
```diff
@@ -77,6 +82,10 @@ public string ToGraph(IGraphStyle style)
             foreach (var transit in transits)
                 dirgraphText += System.Environment.NewLine + transit;
 
+            // Add initial transition if present
+            var initialStateName = initialState.UnderlyingState.ToString();
+            dirgraphText += System.Environment.NewLine + $" init->{initialStateName}[style = \"solid\"]";
```
[the remark this answers is not in the document]
Comment (HenningNT) [reply src/Stateless/Graph/StateGraph.cs]: fixed

**Comment by HenningNT** [comment]
I had completely forgotten about this!
I can add this to 4.4.1, if this still works ;-)

**Comment by ike86** [comment]
# Actual

I've tested it on 8990495
and I get an **actual** output of
```
digraph {
compound=true;
node [shape=Mrecord]
rankdir="LR"
A [label="A"];
B [label="B"];

A -> B [style="solid", label="X"];
 init [label="", shape=point]->A[style = "solid"]
}
```
which produces an ✖ **error**
when I give it to `dot`:
```
syntax error in line 9 near '->'
```

# Expected

After some experimentation I was able to find a format,
which satisfies `dot`:
```
digraph {
compound=true;
node [shape=Mrecord]
rankdir="LR"
A [label="A"];
B [label="B"];
init [label="", shape=point];

A -> B [style="solid", label="X"];
 init->A[style = "solid"]
}
```

I can confirm, that this would result the expected image format as well:
°->A-X->B

File: src/Stateless/Graph/StateGraph.cs [hunk 3]
```diff
@@ -84,7 +84,7 @@ public string ToGraph(IGraphStyle style)
 
             // Add initial transition if present
             var initialStateName = initialState.UnderlyingState.ToString();
-            dirgraphText += System.Environment.NewLine + $" init->{initialStateName}[style = \"solid\"]";
+            dirgraphText += System.Environment.NewLine + $" init [label=\"\", shape=point]->{initialStateName}[style = \"solid\"]";
```
Comment (ike86) [inline src/Stateless/Graph/StateGraph.cs]:
```suggestion
            dirgraphText += System.Environment.NewLine + $" init [label=\"\", shape=point];";
            dirgraphText += System.Environment.NewLine + $" init -> {initialStateName}[style = \"solid\"]";
```

File: test/Stateless.Tests/DotGraphFixture.cs [hunk 4]
```diff
@@ -34,7 +34,7 @@ enum Style
             UML
         }
 
-        static readonly string suffix = Environment.NewLine + $" init->A[style = \"solid\"]" + Environment.NewLine + "}";
+        static readonly string suffix = Environment.NewLine + $" init [label=\"\", shape=point]->A[style = \"solid\"]" + Environment.NewLine + "}";
```
Comment (ike86) [inline test/Stateless.Tests/DotGraphFixture.cs]:
```suggestion
        static readonly string suffix = Environment.NewLine
            + $" init [label=\"\", shape=point];" + Environment.NewLine
            + $" init -> A[style = \"solid\"]" + Environment.NewLine
            + "}";
```

Comment (HenningNT) [reply test/Stateless.Tests/DotGraphFixture.cs]: Thanks!

**Pull request merged**